src/shared/services/path-util.service.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of ng2-json-editor.
+ * Copyright (C) 2016 CERN.
+ *
+ * ng2-json-editor is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * ng2-json-editor is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ng2-json-editor; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+import { Injectable } from '@angular/core';
+
+
+@Injectable()
+export class PathUtilService {
+
+  /**
+   *
+   * @param {Array<any>} path - Element's path
+   * @param {boolean} root - Find nearest or root parent array. True for root and false for nearest array parent
+   * @returns {Array<any>} - Returns the path to the array parent
+   */
+  getNearestOrRootArrayParentInPath(path: Array<any>, root: boolean): Array<any> {
```

Comment:
In the case you mention it is not that bad since the public facing function is clear `addToRootAction`.

But in other cases, it is not very readable, like the public facing ones - https://github.com/inveniosoftware-contrib/ng2-json-editor/pull/100/files#diff-a896742e125e064d80fd8e9c3aa951d5R34 so I would amend those to favor readability of the code. Otherwise one has to be checking the function signature all the time to see what the call is doing.